File: Hola.Shopping.Api.Domain.Model/Product.cs
using System;
using System.Collections.Generic;

namespace Hola.Shopping.Api.Domain.Model
{
    public class Product : IEntity<Guid>
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public Size Size { get; set; }
        public string Color { get; set; }
        public decimal Prize { get; set; }
        public string Description { get; set; }
        public string Reference { get; set; }
        public string Barcode128 { get; set; }
        public bool IsActive { get; set; }
        public Category Category { get; set; }
        public ICollection<ProductOrder> ProductOrders { get; set; }
        public ICollection<Attachment> Attachments { get; set; }
    }
}

File: Client/HolaShopping/src/app/holashopping/models/product/product-list/product.ts
import { BaseEntity } from 'src/app/shared/entities/baseEntity';
import { Guid } from 'guid-typescript';

export class Product extends BaseEntity<Guid> {
    size: string;
    color: string;
    price: string;
    description: string;
    reference: string;
    barcode128: string;
    isActive: boolean;
    category: string;
}
